view/rooms/utils/page.js

```diff
@@ -75,4 +76,63 @@ module.exports = {
       console.log('Cannot inject dynamic contents into iframe.');
     }
   },
+
+  /**
+   * Checks whether a layout object is valid for an array of Bots
+   * @param {Object} layout - Layout object
+   * @param {Array} bots - Bots in the room
+   * @returns {boolean}
+   */
+  botLayoutIsValid(layout, bots) {
+    // Layout does not contain required field
+    if (!layout || !layout.leftColumn || !layout.middleColumn ||
+      !layout.rightColumn) {
+      return false;
+    }
+
+    // Amount of bots in layout does not match amount of bots in room
+    if (layout.leftColumn.length + layout.middleColumn.length +
+      layout.rightColumn.length !== bots.length) {
+      return false;
+    }
+
+    for (const param in layout) {
+      // If a bot in the layout does not actually exist in the room
+      if (layout[param] && layout[param].length
+        && layout[param].every(bot => bots.indexOf(bot) === NEG_ONE)) {
+        return false;
+      }
+    }
+
+    return true;
+  },
+
+  saveLayoutAsCookie() {
```

Comment:
This feels like both a "serialize" function for bot layout and a function to save in cookie. Would it be useful to actually have serialize and deserialize utility functions for bot layout?